prow/plugins/trigger/generic-comment.go

```diff
@@ -147,7 +143,19 @@ type GitHubClient interface {
 // matching cases.
 func FilterPresubmits(honorOkToTest bool, gitHubClient GitHubClient, body string, pr *github.PullRequest, presubmits []config.Presubmit, logger *logrus.Entry) ([]config.Presubmit, []config.Presubmit, error) {
 	org, repo, sha := pr.Base.Repo.Owner.Login, pr.Base.Repo.Name, pr.Head.SHA
-	filter, err := presubmitFilter(honorOkToTest, gitHubClient, body, org, repo, sha, logger)
+	allContexts := sets.String{}
+	failedContexts := sets.String{}
+
+	// We only gather combinedStatus when retest is specified to save GitHub API calls
```

Comment:
We could also have the `PresubmitFilter` accept a `func() (sets.String, error)` that is evaluated to get the list on-demand so we don't have to expose how the re-test works (export the regex etc). This is what we do for the deferred git diff listing for run-if-changed handling

Reply:
done.